src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -109,47 +109,72 @@ def get_comment_contents(issue_list: List[int]) -> List[PullRequestIssue]:
-            continue
-
-        installation = integration_service.get_installation(
-            integration=integration, organization_id=org_id
-        )
+@instrumented_task(name="sentry.tasks.integrations.github_pr_comments")
+def comment_workflow(pr_id):
+    gh_repo_id, pr_key, org_id, issue_list = pr_to_issue_query(pr_id)[0]
+
+    try:
+        organization = Organization.objects.get_from_cache(id=org_id)
+    except Organization.DoesNotExist:
+        logger.error("github.pr_comment.org_missing")
+        return
+
+    # TODO(cathy): add check for OrganizationOption for comment bot
+    if not features.has("organizations:pr-comment-bot", organization):
+        return
+
+    pr_comment = None
+    pr_comment_query = PullRequestComment.objects.filter(pull_request__id=pr_id)
+    if pr_comment_query.exists():
+        pr_comment = pr_comment_query[0]
+
+    try:
+        project = Group.objects.get_from_cache(id=issue_list[0]).project
+    except Group.DoesNotExist:
+        logger.error("github.pr_comment.group_missing", extra={"organization_id": org_id})
+        return
+
+    top_5_issues = get_top_5_issues_by_count(issue_list, project)
+    top_5_issue_ids = [issue["group_id"] for issue in top_5_issues]
+    issue_comment_contents = get_comment_contents(top_5_issue_ids)
+
+    try:
+        repo = Repository.objects.get(id=gh_repo_id)
+    except Repository.DoesNotExist:
+        logger.error("github.pr_comment.repo_missing", extra={"organization_id": org_id})
+        return
+
+    integration = integration_service.get_integration(integration_id=repo.integration_id)
+    if not integration:
+        logger.error("github.pr_comment.integration_missing", extra={"organization_id": org_id})
+        return
+
+    installation = integration_service.get_installation(
+        integration=integration, organization_id=org_id
+    )
+
+    # GitHubAppsClient (GithubClientMixin)
+    client = installation.get_client()
 
-        # GitHubAppsClient (GithubClientMixin)
-        client = installation.get_client()
+    comment_body = format_comment(issue_comment_contents)
 
-        comment_body = format_comment(issue_comment_contents)
+    # client raises ApiError if the request is not successful
+    if pr_comment is None:
+        resp = client.create_comment(repo=repo.name, issue_id=pr_key, data={"body": comment_body})
+
+        current_time = timezone.now()
+        PullRequestComment.objects.create(
+            external_id=resp.body["id"],
+            pull_request_id=pr_id,
+            created_at=current_time,
+            updated_at=current_time,
+            group_ids=top_5_issue_ids,
+        )
+    else:
+        client.update_comment(
+            repo=repo.name, comment_id=pr_comment.external_id, data={"body": comment_body}
+        )
 
-        client.create_comment(repo=repo.name, issue_id=pr_key, data={"body": comment_body})
+        pr_comment.updated_at = timezone.now()
+        pr_comment.group_ids = top_5_issue_ids
```

Comment:
So ideally, we would store all issues associated with the PR in this list. That way, once another event is ingested, we'd only create or update a gh comment for if the issue is new.
We limit it to 24 because that's the P99 for issues per PR. Beyond that there are prs with 20-30k issues, in which case we don't want to store all the issues.

The first 24 is important in particular because they represent the 24 oldest issues associated with the PR.